sklearn/ensemble/iforest.py

```diff
@@ -98,30 +107,35 @@ class IsolationForest(BaseBagging):
            anomaly detection." ACM Transactions on Knowledge Discovery from
            Data (TKDD) 6.1 (2012): 3.
     """
-
+    
     def __init__(self,
+                 base_estimator=None,
                  n_estimators=100,
                  max_samples="auto",
                  max_features=1.,
+                 max_depth="auto",
                  bootstrap=False,
                  n_jobs=1,
                  random_state=None,
                  verbose=0):
+
         super(IsolationForest, self).__init__(
-            base_estimator=ExtraTreeRegressor(
-                max_features=1,
-                splitter='random',
-                random_state=random_state),
-            # here above max_features has no links with self.max_features
+            base_estimator,
```

Comment:
In the __init__ method, (Line#109), I am already passing None as the default value of the base estimator . Then why do we need to make it None again ?
Am I missing something ?